### PR TITLE
Use constants.js to sync up backend and frontend constants.

### PR DESCRIPTION
**Pull request by hoangviet1993**
This PR adds constants.js, the necessary backend infrastructure to read its contents and the backend tests.
PTAL @seanlip.

### CODE REVIEW

File: core/utility/constants.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+"""Loads constants for backend use."""
+
+import json
+import os
+
+
+CONSTANTS_JS_FILEPATH = os.path.join('static', 'assets', 'constants.js')
+
+def parse_json_from_js(js_file):
+    """Extracts JSON object from JS file."""
```
[the remark this answers is not in the document]
Comment (hoangviet1993) [reply core/utility/constants.py]: Replaced with complete docstring, providing details about input and output.

File: core/utility/constants.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    text = js_file.read()
+    first_bracket_index = text.find('= {')
+    last_bracket_index = text.rfind('}')
+    json_text = text[first_bracket_index + 2:last_bracket_index + 1]
+    return json.loads(json_text)
+
+
+class Constants(dict):
+    """Transforms dict to object, attributes can be accesed by dot notation."""
```
[the remark this answers is not in the document]
Comment (hoangviet1993) [reply core/utility/constants.py]: Fixed typo.

File: core/utility/constants.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    __getattr__ = dict.__getitem__
+
+
+with open(CONSTANTS_JS_FILEPATH, 'r') as f:
+    constants = Constants(parse_json_from_js(f)) #pylint: disable=invalid-name
```
[the remark this answers is not in the document]
Comment (hoangviet1993) [reply core/utility/constants.py]: Constants renamed to remove pylint rule overriding.

**Comment by hoangviet1993** [comment]
Here is the current coverage. I will try to get to the Travis PR as soon as I humanly can...maybe next week?
screen shot 2018-09-07 at 9 40 21 am

File: core/utility/constants.py [hunk 4]
```diff
@@ -23,7 +23,14 @@
 CONSTANTS_JS_FILEPATH = os.path.join('static', 'assets', 'constants.js')
 
 def parse_json_from_js(js_file):
-    """Extracts JSON object from JS file."""
+    """Extracts JSON object from JS file.
+    Args:
```
Comment (seanlip) [inline core/utility/constants.py]: Add a newline above this line

File: core/utility/constants.py [hunk 5]
```diff
@@ -23,7 +23,14 @@
...
+        js_file: file. A stream handling object of constants.js.
```
Comment (seanlip) [inline core/utility/constants.py]: "handling" is a bit woolly. Maybe: "A stream representing the constants.js file" or similar?

File: core/utility/constants.py [hunk 6]
```diff
@@ -23,7 +23,14 @@
...
+
+    Returns:
+        dict(str, str). The dictionary containing variable name as key and
```
Comment (seanlip) [inline core/utility/constants.py]: Why do you need the last part (unicode strings)? I thought we could represent numbers, booleans, even null in this JSON representation.

(If that's the case, then something like "A dict mapping constant names to their values." might be better.)

File: core/utility/constants.py (hunk 4 above)
Comment (hoangviet1993) [reply core/utility/constants.py]: Done!

File: core/utility/constants.py (hunk 5 above)
Comment (hoangviet1993) [reply core/utility/constants.py]: Done!

File: core/utility/constants.py (hunk 6 above)
Comment (hoangviet1993) [reply core/utility/constants.py]: So I was getting `u'abcdef'` when I print out the dict values. Anyhow, I re-word the docstring.

**Comment by hoangviet1993** [comment]
PTAL @seanlip

**Comment by seanlip** [comment]
LGTM. Thanks!

**Pull request merged**